packages/adapter-components/src/deployment/deploy/graph.ts

```diff
@@ -92,7 +92,7 @@ export const createDependencyGraph = <ClientOptions extends string, AdditionalAc
       })
     })
     defQuery.query(typeName)?.actionDependencies?.forEach(({ first, second }) => {
-      graph.addEdge(toNodeID(typeName, second), toNodeID(typeName, first))
+      graph.addEdge(toNodeID(typeName, second), toNodeID(typeName, first), { createMissingNodes: false })
```

Comment:
I think this logic is problematic, because if we have a << b << c and b is missing, we will lose the implied dependency a << c.
wdyt about instead ensuring there are nodes, e.g.
```suggestion
      // add empty nodes if they don't already exist
      [first, second].forEach(n => {
        const id = toNodeID(typeName, n)
        if (!graph.has(id)) {
          graph.addNode(id, [], {
            action: n as ActionName | AdditionalAction,
            typeActionChanges: [],
            typeName,
          })
        }
      })
      graph.addEdge(toNodeID(typeName, second), toNodeID(typeName, first))
```
(and not changing the dag)?

Reply:
I'm honestly not sure we should fix this, as it likely hides a development error. can throw a clearer error instead...